packages/app-mobile/components/NoteEditor/EditLinkDialog.tsx

```diff
@@ -0,0 +1,156 @@
+// Dialog allowing the user to update/create links
+
+const React = require('react');
+const { useState, useEffect, useMemo, useRef } = require('react');
+const { StyleSheet } = require('react-native');
+const { View, Modal, Text, TextInput, Button } = require('react-native');
+
+import { themeStyle } from '@joplin/lib/theme';
+import { _ } from '@joplin/lib/locale';
+import { EditorControl } from './types';
+import SelectionFormatting from './SelectionFormatting';
+
+interface LinkDialogProps {
+	editorControl: EditorControl;
+	selectionState: SelectionFormatting;
+	visible: boolean;
+	themeId: number;
+}
+
+const EditLinkDialog = (props: LinkDialogProps) => {
+	// The content of the link selected in the editor (if any)
+	const editorLinkData = props.selectionState.linkData;
+	const [linkLabel, setLinkLabel] = useState('');
+	const [linkURL, setLinkURL] = useState('');
+
+	// Reset the label and URL when shown/hidden
+	useEffect(() => {
+		setLinkLabel(editorLinkData.linkText ?? props.selectionState.selectedText);
+		setLinkURL(editorLinkData.linkURL ?? '');
+	}, [props.visible]);
+
+	// Updates the content of the CodeMirror editor based on internal state.
+	const updateEditor = () => {
+		props.editorControl.updateLink(linkLabel, linkURL);
+	};
+
+	const [styles, placeholderColor] = useMemo(() => {
+		const theme = themeStyle(props.themeId);
+
+		const styleSheet = StyleSheet.create({
+			modalContent: {
+				margin: 15,
+				padding: 30,
+				backgroundColor: theme.backgroundColor,
+
+				elevation: 5,
+				shadowOffset: {
+					width: 1,
+					height: 1,
+				},
+				shadowOpacity: 0.4,
+				shadowRadius: 1,
+			},
+			button: {
+				color: theme.color2,
+				backgroundColor: theme.backgroundColor2,
+			},
+			text: {
+				color: theme.color,
+			},
+			header: {
+				color: theme.color,
+				fontSize: 22,
+			},
+			input: {
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+
+				minHeight: 48,
+				borderBottomColor: theme.backgroundColor3,
+				borderBottomWidth: 1,
+			},
+			inputContainer: {
+				flexDirection: 'column',
+				paddingBottom: 10,
+			},
+		});
+		const placeholderColor = theme.colorFaded;
+		return [styleSheet, placeholderColor];
+	}, [props.themeId]);
+
+	const submit = () => {
+		updateEditor();
+		props.editorControl.hideLinkDialog();
+	};
```

Comment:
All function that are used in callbacks should be defined using `useCallback` to avoid stale data, and used variables should be passed in the dependencies array. Also please name it onSomething (here onSubmit) to make it clear it's a callback. Maybe we need some eslint rule for this.